src/masternodeman.cpp

```diff
@@ -856,8 +857,7 @@ void CMasternodeMan::CheckSameAddr()
                     pverifiedMasternode = pmn;
                 }
             } else {
-                // clear the reference
-                pverifiedMasternode = NULL;
+                pverifiedMasternode = pmn->IsPoSeVerified() ? pmn : NULL;
             }
             pprevMasternode = pmn;
         }
```

Comment:
This doesn't ensure that all but 1 MN will get banned.

I think you actually want something simpler, like this: 
https://gist.github.com/tgflynn/47b19dfe844c372aa7a97b0cca453ec4